### PR TITLE
Feature/food items/edit item

### LINKED ISSUE #6: As a user, I want to edit a food item
"Oh no, I made a mistake. I need to update my item!"

- backend
  - PUT endpoint `/api/food/{id}`
  - user must be logged in and can only edit their own posts
- frontend
  - form with all details already filled in
  - route: `/food/{id}/edit`
  - link from item list (only on the user's own items)
  - another link from item details page (only on the user's own items)
- unit and integration tests

### PR DESCRIPTION
**Pull request by valfink**
Resolve #6

### CODE REVIEW

File: backend/src/main/java/com/github/valfink/backend/fooditem/FoodItemService.java [hunk 1]
```diff
@@ -85,7 +96,37 @@ public FoodItemDTOResponse addFoodItem(FoodItemDTORequest foodItemDTORequest, Mu
...
+                foodItemDTORequest.title(),
+                photoUri,
+                foodItemDTORequest.location(),
+                foodItemDTORequest.pickupUntil(),
+                foodItemDTORequest.consumeUntil(),
+                foodItemDTORequest.description()
+        ));
 
         return foodItemDTOResponseFromFoodItem(foodItem);
     }
```
Comment (timlenny) [inline backend/src/main/java/com/github/valfink/backend/fooditem/FoodItemService.java:132]: The code in the class is very good. If more coding is added here in the future, you could create another helper class

File: frontend/src/util/foodItemRequests.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import {FoodItemFormData} from "../model/FoodItemFormData";
+
+const API_URL = "/api/food";
+
+export async function fetchAllFoodItems(setAppIsLoading: React.Dispatch<React.SetStateAction<number>>) {
+    let foodItems: FoodItem[] = [];
+
+    setAppIsLoading(oldValue => oldValue++);
+    try {
+        const response = await axios.get(API_URL);
```
Comment (timlenny) [inline frontend/src/util/foodItemRequests.ts:13]: Possibly integrate catch / finally directly into the axios call

**Review by timlenny — APPROVED** [review]
Very good code. I didn't really find anything negative. Good job! 👍

File: frontend/src/util/foodItemRequests.ts (hunk 2 above)
Comment (valfink) [reply frontend/src/util/foodItemRequests.ts:13]: Is is the same code as `axios.get... .then .catch .finally` - just written as an `async` function :)

**Pull request merged**